### ISSUE TITLE
External editor changes being resetted in vi mode

### ISSUE THREAD

**Issue by ShinyZero0** [nushell/nushell#9014]
### Describe the bug

It happens only if you edit the command from history

### How to reproduce

1. Get some command from history with up arrow
2. Open editor with Ctrl-O
3. Edit something, save and exit
4. Switch vi mode to insert or to normal and back to insert (i did it with `a`)
5. Changes are resetted so that command is the same as it was in history

### Expected behavior

Changes to be saved

### Configuration

| key                | value                                                             |
| ------------------ | ----------------------------------------------------------------- |
| version            | 0.78.0                                                            |
| branch             |                                                                   |
| commit_hash        |                                                                   |
| build_os           | linux-x86_64                                                      |
| build_target       | x86_64-unknown-linux-gnu                                          |
| rust_version       | rustc 1.68.2 (9eb3afe9e 2023-03-27) (built from a source tarball) |
| cargo_version      | cargo 1.68.1                                                      |
| build_time         | 1980-01-01 00:00:00 +00:00                                        |
| build_rust_channel | release                                                           |
| features           | default, zip                                                      |
| installed_plugins  |                                                                   |

**Comment by amtoine** [comment]
i'm not sure i'm able to reproduce this on latest `main` :confused: 

is the behaviour you see reliable?

**Comment by ShinyZero0** [comment]
Yes, i just even tried that with nano instead of neovim. Probably that was fixed as i'm on 0.78

**Comment by amtoine** [comment]
yeah maybe, i've never see this behaviour, that would be amazing if you could try it on the new release when you have time? :relieved:

**Comment by ShinyZero0** [comment]
Yes, just reproduced on 79. Maybe something wrong with [my config](https://github.com/ShinyZero0/dotfiles/blob/void/.config/nushell/config.nu), i just commented out all usings and sources to get vi mode and reproduced this bug

**Comment by sholderbach** [comment]
Thanks for including the config! Don't see anything particularly special which would be immediately suspect 🤔 

> Yes, just reproduced on 79. Maybe something wrong with [my config](https://github.com/ShinyZero0/dotfiles/blob/void/.config/nushell/config.nu?rgh-link-date=2023-04-26T17%3A46%3A33Z), i just commented out all usings and sources to get vi mode and reproduced this bug
